accord-core/src/main/java/accord/topology/TopologyManager.java

```diff
@@ -587,7 +590,9 @@ public Topologies preciseEpochs(Unseekables<?> select, long minEpoch, long maxEp
     {
         Epochs snapshot = epochs;
 
-        TopologyMismatch tm = TopologyMismatch.checkForMismatch(snapshot.get(maxEpoch).global(), select);
+        EpochState maxState = snapshot.get(maxEpoch);
+        Invariants.checkState(maxState != null, "Unable to find epoch %d; known epochs are %d -> %d", maxEpoch, snapshot.minEpoch(), snapshot.currentEpoch);
```

Comment:
in C* we bootstrap with epoch=9 but there are pending txn in epoch=8... this means that we are failing to handle those txn and SimpleProgressLog starts spamming us... I tried to look at how to add historic epochs but it was getting a bit annoying and didn't actually impact the bootstrap tests... so figured I could delay that problem to another patch